MobileSyncExplorerKotlinTemplate/app/src/main/java/com/salesforce/mobilesyncexplorerkotlintemplate/core/salesforceobject/JSONObject+Coerce.kt

```diff
@@ -0,0 +1,89 @@
+package com.salesforce.mobilesyncexplorerkotlintemplate.core.salesforceobject
+
+import com.salesforce.androidsdk.mobilesync.target.SyncTarget.*
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+@Throws(CoerceException::class)
+fun JSONObject.getRequiredStringOrThrow(key: String, valueCanBeBlank: Boolean = true): String =
```

Comment:
Only two of the methods in this class is actually used.

Reply:
This is actually a good discussion point:  For tightly-related functionality I generally like having all use cases implemented all at once in case I need them.  In my opinion it makes development easier because one does not have to question why related functions weren't implemented to begin with.

Would it be okay to leave them, or is it unhelpful noise to consumers of the template?

Reply:
One of the usual goals of a template app is typically to be as simple as possible (sometime even to the point of being barebones) to make it as easy to understand as quickly as possible.  I think there is quite a bit to digest in this app.